src/FileIO.h

```diff
@@ -60,4 +61,26 @@ struct ScopedFileDescriptor {
       close(fd);
    }
 };
+
+// Do file access as root user for the function passed in.
+// Example: auto result = FileIO::SudoFile(FileIO::ReadAsciiFileContent, filePath);
```

Comment:
I don't want to go into nitpicking mode so I think it's fine. I'll wait to plus it till I've tested it but review wise it's fine 

(If Alex has tested it already then I'm good with merging)